org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/PDEMavenBundlePluginConfigurator.java

```diff
@@ -53,93 +50,85 @@
-				MavenProblemInfo problem = new MavenProblemInfo(
-						"There is currently no execution that generates a manifest, consider adding an execution for one of the following goal: "
-								+ (isFelix(execution.getPlugin()) ? FELIX_MANIFEST_GOAL
-										: Arrays.toString(BND_MANIFEST_GOALS))
-								+ ".",
-						IMarker.SEVERITY_WARNING, location);
-				problems.add(problem);
-				break;
-			}
+		if (!hasManifestExecution && !executions.isEmpty()) {
+			MojoExecution execution = executions.get(0);
+			createWarningMarker(request, execution, "executions",
+					"There is currently no execution that generates a manifest, consider adding an execution for one of the following goal: "
+							+ (isFelix(execution.getPlugin()) ? FELIX_MANIFEST_GOAL : BND_MANIFEST_GOALS) + ".");
 		}
-		if (problems.size() > 0) {
-			this.markerManager.addErrorMarkers(request.getPom(), IMavenConstants.MARKER_LIFECYCLEMAPPING_ID, problems);
-		}
-		IProject project = request.getProject();
-		IMavenProjectFacade facade = request.getMavenProjectFacade();
 
+		IMavenProjectFacade facade = request.getMavenProjectFacade();
 		IPath metainfPath = getMetainfPath(facade, executions, monitor);
+		PDEProjectHelper.addPDENature(facade.getProject(), metainfPath, monitor);
+	}
 
-		PDEProjectHelper.addPDENature(project, metainfPath, monitor);
+	private void createWarningMarker(ProjectConfigurationRequest request, MojoExecution execution, String attribute,
+			String message) {
+		SourceLocation location = SourceLocationHelper.findLocation(execution.getPlugin(), attribute);
+
+		String[] gav = location.getResourceId().split(":");
+		IMavenProjectFacade facade = projectManager.getMavenProject(gav[0], gav[1], gav[2]);
+		if (facade == null) {
+			// attribute specifying project (probably parent) is not in the workspace.
+			// The following code returns the location of the project's parent-element.
+			location = SourceLocationHelper.findLocation(request.getMavenProject(), new MojoExecutionKey(execution));
+			facade = request.getMavenProjectFacade();
+		}
+		MavenProblemInfo problem = new MavenProblemInfo(message, IMarker.SEVERITY_WARNING, location);
+		markerManager.addErrorMarker(facade.getPom(), IMavenConstants.MARKER_LIFECYCLEMAPPING_ID, problem);
 	}
 
-	private boolean isFelixBundleGoal(MojoExecution execution) {
+	private boolean isFelixManifestGoal(MojoExecution execution) {
 		return FELIX_MANIFEST_GOAL.equals(execution.getGoal());
 	}
 
 	private boolean isBNDBundleGoal(MojoExecution execution) {
-		String executionGoal = execution.getGoal();
-		for (String goal : BND_MANIFEST_GOALS) {
-			if (goal.equals(executionGoal)) {
-				return true;
-			}
-		}
-		return false;
+		return BND_MANIFEST_GOALS.contains(execution.getGoal());
 	}
 
 	@Override
 	public void configureClasspath(IMavenProjectFacade facade, IClasspathDescriptor classpath, IProgressMonitor monitor)
-			throws CoreException {
+			throws CoreException { // nothing to do
```

Comment:
PDEProjectHelper has some code that seems to handle class path changes, do you think we should call it here?

Reply:
I'm not yet sure about that, but I will check it as soon as I continue my work in this area!